contracts/Comptroller.sol

```diff
@@ -1150,151 +1084,196 @@ contract Comptroller is ComptrollerV4Storage, ComptrollerInterface, ComptrollerE
+     * @param cToken The market in which the holder is interacting
+     * @param holder The address of the holder to distribute COMP to
+     * @param distribute Whether to distribute accrued COMP
+     * @param marketBorrowIndex The given market's borrow index
+     * @param marketState Supply or borrow state for the market (depending on isSupply)
+     * @param vestingState Supply or borrow vesting state for the market (depending on isSupply)
+     * @param isSupply True if distributeMarketComp called by distributeSupplierComp
+     */
+    function distributeMarketComp(address cToken, address holder, bool distribute, CompMarketState storage marketState, CompMarketState storage vestingState, bool isSupply,
+            Exp memory marketBorrowIndex) internal {
+        Double memory marketIndex = Double({mantissa: marketState.index});
+        Double memory holderIndex;
+        if (isSupply) {
+            holderIndex = Double({mantissa: compSupplierIndex[cToken][holder]});
+            compSupplierIndex[cToken][holder] = marketIndex.mantissa;
+
+            if (holderIndex.mantissa == 0 && marketIndex.mantissa > 0) {
+                holderIndex.mantissa = compInitialIndex;
+            }
+        } else {
+            holderIndex = Double({mantissa: compBorrowerIndex[cToken][holder]});
+            compBorrowerIndex[cToken][holder] = marketIndex.mantissa;
+        }
+
+        // Accrue vested COMP
+        uint holderAccrued = compAccrued[holder];
+        if (lastVestingBlock > vestingBlock[holder]) {
+            vestingBlock[holder] = lastVestingBlock;
+
+            holderAccrued = add_(holderAccrued, compVesting[holder]);
+            compVesting[holder] = 0;
+        }
+
+        Double memory marketVestingIndex = holderIndex;
+        if (vestingState.index > holderIndex.mantissa) {
+            marketVestingIndex = Double({mantissa: vestingState.index});
+        }
+
+        if (isSupply || holderIndex.mantissa > 0) {
+            // Accrue COMP that was earned leading up to vesting event
+            Double memory deltaIndex;
+            uint holderDelta;
+            uint holderHoldings;
+            if (isSupply) {
+                holderHoldings = CToken(cToken).balanceOf(holder);
+            } else {
+                holderHoldings = div_(CToken(cToken).borrowBalanceStored(holder), marketBorrowIndex);
+            }
+
+            if (marketVestingIndex.mantissa > holderIndex.mantissa) {
+                deltaIndex = sub_(marketVestingIndex, holderIndex);
+                holderDelta = mul_(holderHoldings, deltaIndex);
+                holderAccrued = add_(holderAccrued, holderDelta);
+            }
+
+            // Vest any new COMP earned after vesting event
+            deltaIndex = sub_(marketIndex, marketVestingIndex);
+            holderDelta = mul_(holderHoldings, deltaIndex);
+            compVesting[holder] = add_(compVesting[holder], holderDelta);
```

Comment:
Same here, see my earlier comment.

Reply:
Same principle applies here, we are using `deltaIndex` and `holderDelta` as throwaway variables to calculate `holderAccrued`!